### PR TITLE
Add Template Classes.

### PR DESCRIPTION
**Pull request by jessetilro**
Added a bunch of templates for the classes described in the first version of the UML class diagram designed for Sprint 2. Also introduced templates for corresponding test suites using parallel class hierarchy.

The few tiny features and integrations included are not tested for they will quite likely change a lot during this sprint. In addition some static errors have slipped through, because of the classes being templates rather than full implementations.

The creation of this relatively large bunch of template classes was agreed upon in order to allow cooperation to be done more effectively, for the templates eliminate some coordination overhead.

### CODE REVIEW

**Comment by wouterz** [comment]
LGTM (indentation is goed, voor nu)

File: src/main/java/pixelperfect/spaceship/Route.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  /**
+   * Randomly generate a Route (which is an aggregation of RouteNodes) using the random
+   * RouteGenerator.
+   * 
+   * @return A randomly generated route.
+   */
+  public static Route generateRoute() {
+    return new Route(42);
```
Comment (FlorisXll) [inline src/main/java/pixelperfect/spaceship/Route.java]: Found you!

Comment (FlorisXll) [inline src/main/java/pixelperfect/spaceship/Route.java]: Also, we can just give a RouteGenerator object as private atribute of the class and initiate it when a Route is created. 
Then I would like to add the fact that this class should be a Singleton: only one instance of the class can be/should be alive at the time.

**Comment by FlorisXll** [comment]
Why is the group of Route classes in the spaceship package?

File: src/main/java/pixelperfect/spaceship/Route.java (hunk 1 above)
Comment (jessetilro) [inline src/main/java/pixelperfect/spaceship/Route.java]: Singleton seems like a good idea to me.

**Comment by jessetilro** [comment]
I suggest you make a PR for refactoring the packages.

**Pull request merged**